routes/[version]/apikey.js

```diff
@@ -0,0 +1,55 @@
+const { checkAuth } = require('../../middlewares/auth');
+const { selectAPIToken, setAPIToken, removeAPIToken } = require('../../services/db');
+
+module.exports = {
+    get: [checkAuth, get],
+    put: [checkAuth, put],
+    delete: [checkAuth, del],
+};
+
+function generateAPIToken() {
+    let result = '',
+        counter = 0;
+    const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_';
+    const charlen = chars.length;
+
+    while (counter < 11) {
+        result += chars.charAt(Math.floor(Math.random() * charlen));
+        counter += 1;
+    }
+    return 'METIS' + result;
+}
+
+/**
+ * @api {put} /v0/apikey Create new API key
+ * @apiName SetAPIKey
+ * @apiGroup Users
+ * @apiDescription Create new API key
+ * @apiPermission GUI_ONLY
+ */
+async function put(req, res, next) {
+    return res.json({ apikey: await setAPIToken(req.user.id, generateAPIToken()) });
+}
+
+/**
+ * @api {del} /v0/apikey Remove API key
+ * @apiName RemoveAPIKey
+ * @apiGroup Users
+ * @apiDescription Remove all API keys
+ * @apiPermission GUI_ONLY
+ */
+async function del(req, res, next) {
+    removeAPIToken(req.user.id);
+    return res.json({});
```

Comment:
Maybe `{ apikey: null }` with response code 200 or just nothing with response code 204? I don't like the fact that the data structure is changing.

Reply:
I vote for 204 code, more info [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE)

A [202](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202) (Accepted) status code if the action will likely succeed but has not yet been enacted.
A [204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) (No Content) status code if the action has been enacted and no further information is to be supplied.
A [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) (OK) status code if the action has been enacted and the response message includes a representation describing the status.